test/functional/mempool_package_limits.py

```diff
@@ -3,20 +3,41 @@
 # Distributed under the MIT software license, see the accompanying
 # file COPYING or http://www.opensource.org/licenses/mit-license.php.
 """Test logic for limiting mempool and package ancestors/descendants."""
-
-from decimal import Decimal
-
 from test_framework.blocktools import COINBASE_MATURITY
-from test_framework.test_framework import BitcoinTestFramework
 from test_framework.messages import (
-    COIN,
     WITNESS_SCALE_FACTOR,
 )
+from test_framework.test_framework import BitcoinTestFramework
 from test_framework.util import (
     assert_equal,
 )
 from test_framework.wallet import MiniWallet
 
+# Decorator to
+# 1) check that mempool is empty at the start of a subtest
+# 2) run the subtest, which may submit some transaction(s) to the mempool and
+#    create a list of hex transactions
+# 3) testmempoolaccept the package hex and check that it fails with the error
+#    "package-mempool-limits" for each tx
+# 4) after mining a block, clearing the pre-submitted transactions from mempool,
+#    check that submitting the created package succeeds
+def check_package_limits(func):
```

Comment:
Nice :100: 
Just a couple of nit minor suggestions to improve the readability of the code:

    Add a blank line after the first assertion assert_equal(0, node.getmempoolinfo()["size"])
    Add a blank line after the line where package_hex is assigned to func(self, *args, **kwargs)

Separating the operations as written in the decorator comment, also goes well with PEP9 guidelines.